### PR TITLE
feat: implement the nested object and array in the Table Node

### PR DESCRIPTION
**Pull request by stone-lyl**
feat: implement the nested object in the Table Node
feat: implement the nested array in the Table Node
test: add the nested object test in ItemCollection
test: add the nested array test

before:
image

after: 
image

### CODE REVIEW

File: packages/ui/src/components/Node/ItemCollection.test.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    ];
+
+    const { headers, rows } = new ItemCollection(mockData).toTable();
+    expect(headers).toMatchInlineSnapshot(`
+      [
+        "foo",
+        "baz",
+      ]
+    `);
+    expect(rows).toMatchInlineSnapshot(`
```
Comment (ajthinking) [inline packages/ui/src/components/Node/ItemCollection.test.ts:26]: In general, and in this case, are there reasons to use snapshot over [toMatchObject](https://vitest.dev/api/expect#tomatchobject)? We loose IDE help when its just a string so I think prefer toMatchObject

File: packages/ui/src/components/Node/ItemCollection.ts [hunk 2]
```diff
@@ -1,26 +1,83 @@
...
+  constructor(public items: ItemValue[]) {
+  }
+
+  private serializeArrayContent(arrayContent: unknown[]): string {
+    return JSON.stringify(arrayContent).replace(/\n/g, '');
   }
-}
\ No newline at end of file
+
+  public toTable() {
```
Comment (ajthinking) [inline packages/ui/src/components/Node/ItemCollection.ts:13]: Public is infered by default, so we can leave this out. Do keep public methods above private ones

File: packages/ui/src/components/Node/ItemCollection.ts [hunk 3]
```diff
@@ -1,26 +1,83 @@
...
+    const headers: Set<string> = new Set();
```
Comment (ajthinking) [inline packages/ui/src/components/Node/ItemCollection.ts:14]: Using a Set from scratch, nice 👍

File: packages/ui/src/components/Node/ItemCollection.ts [hunk 4]
```diff
@@ -1,26 +1,83 @@
...
+    this.items.forEach(item => {
+      if (typeof item === 'object' && item !== null && !Array.isArray(item)) {
+        buildHeaders(item);
+      }
+    });
+
+    /**
+     * @description get value by header's path
+     */
+    const getValueByPath = (object: ItemValue, path: string[]): string | undefined => {
```
Comment (ajthinking) [inline packages/ui/src/components/Node/ItemCollection.ts:43]: Not sure if usable here, but can be good to know this utility exists: packages/core/src/utils/get.ts

**Review by ajthinking — APPROVED** [review]
Good stuff 👍

File: packages/ui/src/components/Node/ItemCollection.test.ts (hunk 1 above)
Comment (stone-lyl) [reply packages/ui/src/components/Node/ItemCollection.test.ts:26]: This case passes: `expect({foo:'hi', bar: 'hello'}).toMatchObject({bar: 'hello'});` 

using `toMatchObject` when we're only interested in checking if certain properties of the object match our expectations. 
 In this scenario, I'm looking to test if the entire result meets the expectations.

File: packages/ui/src/components/Node/ItemCollection.ts (hunk 4 above)
Comment (stone-lyl) [reply packages/ui/src/components/Node/ItemCollection.ts:43]: `getValueByPath` includes some special functionalities for handling types of array and object values

File: packages/ui/src/components/Node/ItemCollection.test.ts (hunk 1 above)
Comment (ajthinking) [reply packages/ui/src/components/Node/ItemCollection.test.ts:26]: Got it! Can we use toEqual for those cases? 

```ts
  it('can use toEqual even on nested objects', () => {
    const a = { foo: { bar: { baz: 'qux' } } };
    const b = { foo: { bar: { baz: 'qux' } } };
    expect(a).toEqual(b);
  });
```
This passes

File: packages/ui/src/components/Node/ItemCollection.ts (hunk 4 above)
Comment (ajthinking) [reply packages/ui/src/components/Node/ItemCollection.ts:39]: 👍

File: packages/ui/src/components/Node/ItemCollection.ts (hunk 2 above)
Comment (stone-lyl) [reply packages/ui/src/components/Node/ItemCollection.ts]: 🆗

File: packages/ui/src/components/Node/ItemCollection.test.ts (hunk 1 above)
Comment (ajthinking) [reply packages/ui/src/components/Node/ItemCollection.test.ts]: Seeing the update👍 No downside with this approach right? Lets merge when you are ready

**Pull request merged**